### ISSUE TITLE
Plugins configuration/state is not correctly set on maps created with contexts 

### ISSUE THREAD

**Issue by ale-cristofori** [geosolutions-it/MapStore2#9097]
## Description
When creating maps with contexts, having the save plugin, and other plugins that can be enabled or disabled. Such plugins won't retain their state across page reloads. e.g. if a map is saved with the identify plugin disabled, when refreshing the same map, after the refresh the plugin will wrongly enable. 

## How to reproduce

-  
- 

*Expected Result*

*Current Result*

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

**Comment by tdipisa** [comment]
@ale-cristofori I don't think it is a bug. The behavior you are mentioning is not part of [the original ACs](https://github.com/geosolutions-it/MapStore2/issues/4722) therefore it need to be implemented as a enhancement.

**Comment by ale-cristofori** [comment]
Client added more cases where they related the bug might take place


_Some extra information: I see the same behavior with the WMF Search Services config of the search bar._ 

_The following map has search services configured:_
_https://mapstore-tni.gis.cloud.mow.vlaanderen.be/mapstore/#/context/tandem/3973_

_When you refresh the map, the configuration is gone. When you load it into another context (eg. https://mapstore-tni.gis.cloud.mow.vlaanderen.be/mapstore/#/context/tandem_embed/3973),_ _the configuration is back again. When you refresh the web page, the configuration is gone again..._ _Very similar behavior as with the map details used together with contexts as described in the first comment._

**Comment by tdipisa** [comment]
@ale-cristofori I spent some time checking about the purpose of this issue and I think we should clarify a couple of things.

The state of plugins that can be enabled on the fly from the MS UI through the tool User Extensions is not expected to be persisted when the map is saved. That's by design and I'm not sure at the moment it would be good to make this available 
because this will imply more code and more combinations to be maintained and tested with possible regressions or conflicts with the configuration imposed by the context that can change over time.
At the moment there is anyway a support, provided by the User Session plugin, that could be used to provide requested in this issue. Basically the User Session, if enabled for a context, allows by default to keep the current map state of the 
user's session in the browser localStorage. The core APIs already allow to extend this for other part of the state, like effectively the activation state of a plugin; it has already been done for some downstream projects and should require as small 
development as far as I remember (I suppose it also depends on which part of the state you want to manage within the User Session plugin umbrella).

Therefore I would be good in my opinion to legerave on the User Session in my opinion for the reason explained above.


As for the following

> Some extra information: I see the same behavior with the WMF Search Services config of the search bar.

> The following map has search services configured:
https://mapstore-tni.gis.cloud.mow.vlaanderen.be/mapstore/#/context/tandem/3973

That context doesn't seem to have any custom search services configured. Only the nominatim one is present that's anyway available by default.

> When you refresh the map, the configuration is gone. When you load it into another context (eg. https://mapstore-tni.gis.cloud.mow.vlaanderen.be/mapstore/#/context/tandem_embed/3973), the configuration is back again. When you refresh the web page, the configuration is gone again... Very similar behavior as with the map details used together with contexts as described in the first comment.

Due to also the fact that no custom search services are configured in that contec, this use case should be clarified also because there are different ways to configure a search service:

1. Directly from the context wizard (step 3 configuration of the Search plugin) whers the admin can define default search services always present in the context. They don't appear in the Search tool UI of the context viewer, because they can 
be configured only by the administrator in the wizard, but they are effectively used there if the user perform a search

2. From the UI of the context viewer where, if the Search plugin is defined, it is possible for users to define their additional custom search services  

The same I've explained above is anyway valid and that could be applied (for point 2 right above) to new custom services configured by the user through the UI of the context viewer. 
The possibility to include this in what managed by the User Session plugin should be anyway checked and evaluated.

Effectively, it is anyway already possible for the user to configure some custom services through the UI of the context viewer and then save a new map from the context to persist there the state of created search services.

**Comment by ale-cristofori** [comment]
I am ok for using the user session. As for the investigation and the configuration of the search tool

@tdipisa as for these things that are not reprpducible, I reproduced them on DEV
on [this context ](https://dev-mapstore.geosolutionsgroup.com/mapstore/#/context/Issue_9097)
and this map
_Directly from the context wizard (step 3, configuration of the Search plugin) where the admin can define default search services_ _always present in the context. They don't appear in the Search tool UI of the context viewer, because they can_
_be configured only by the administrator in the wizard, but they are effectively used there if the user perform a search_

this does not seem to be the case on DEV, see the video below on this context 

https://github.com/geosolutions-it/MapStore2/assets/14953970/a305ae62-4eb0-43f7-8cf8-b31f4c098d07

**Comment by tdipisa** [comment]
@ale-cristofori it seems the configuration you are using is not correct. See what configured [here] (https://dev-mapstore.geosolutionsgroup.com/mapstore/#/context/tdipisa_demo_theme_imported) to see how to.

**Comment by ale-cristofori** [comment]
@tdipisa and @offtherailz

Taking away the part about the configuration of the search services. Within this ticket we have now to focus on the following requirements 

Persisting the configuration of the `PhotoViewer` extension both on the maps and contexts resources, the configurations to keep are the ones shown in this plugin configuration window
![image](https://github.com/geosolutions-it/MapStore2/assets/14953970/424b4b1d-dfe6-4994-b52f-95c971538ce4)

As discussed with @offtherailz in phase of estimation of the ticket **2.5 days** we agreed that one method to achieve the save requirement (**only for the PhotoViewer extension**) would be to implement the state of the plugin directly in the map configuration. Just restricting this functionality to the PhotoViewer plugin. The plugin state, including it activation/de-activation, its configuration in terms of layers URL, point layer name, default zoom level, etc will be retained to the last time the user clicks on the save button. Conversely if a user on a map configures the plugin and changes page without saving the map, the configurations won't be retained.

The fix will be implemented on the client downstream project repo for both MapStore and the photoviewer extension, both on client's codeCommit AWS version management.

- The name of the MapStore downstream project repo is **cc-dmow-mapstore-config** the branch to start develop upon is **dev** which points to the core version of MapStore **2023.01.02**, the name of the PhotoViewer extension repo is **cc-dmow-fietsgis-temp** and we need to start develop on the **master** branch.
